include/ck/host_utility/flush_cache.hpp

```diff
@@ -0,0 +1,231 @@
+            }
+
+            {
+                void* pBDeviceBuf;
+                hip_check_error(hipMalloc(static_cast<void**>(&pBDeviceBuf), size_b_));
+                hip_check_error(hipMemcpy(static_cast<void*>(pBDeviceBuf),
+                                          const_cast<void*>(p_b_grids[0]),
+                                          size_b_,
+                                          hipMemcpyDeviceToDevice));
+                p_b_grids.push_back(pBDeviceBuf);
+            }
+
+            {
+                void* pCDeviceBuf;
+                hip_check_error(hipMalloc(static_cast<void**>(&pCDeviceBuf), size_c_));
+                p_c_grids.push_back(pCDeviceBuf);
+            }
+        }
+    }
+
+    void Next()
+    {
+        if(rotating_count > 1)
+        {
+            int idx      = iter++ % rotating_count;
+            arg.p_a_grid = reinterpret_cast<ADataType>(p_a_grids[idx]);
+            arg.p_b_grid = reinterpret_cast<BDataType>(p_b_grids[idx]);
+            arg.p_c_grid = reinterpret_cast<CDataType>(p_c_grids[idx]);
+        }
+    }
+    void Print()
+    {
+        std::cout << "RotatingMemWrapper{" << size_a << "," << size_b << "," << size_c
+                  << "} rotating_count: " << rotating_count << std::endl;
+    }
+    ~RotatingMemWrapper()
+    {
+        if(rotating_count > 1)
+        {
+            // restore ptr
+            arg.p_a_grid = reinterpret_cast<ADataType>(p_a_grids[0]);
+            arg.p_b_grid = reinterpret_cast<BDataType>(p_b_grids[0]);
+            arg.p_c_grid = reinterpret_cast<CDataType>(p_c_grids[0]);
+#if 0
+            // test rotating buffer gemm result
+            hip_check_error(hipMemcpy(static_cast<void*>(p_c_grids[0]),
+                                      const_cast<void*>(p_c_grids[--iter % rotating_count]),
+                                      size_c,
+                                      hipMemcpyDeviceToDevice));
+#endif
+            // free device mem
+            for(size_t i = 1; i < rotating_count; i++)
+            {
+                hip_check_error(hipFree(const_cast<void*>(p_a_grids[i])));
+                hip_check_error(hipFree(const_cast<void*>(p_b_grids[i])));
+                hip_check_error(hipFree(p_c_grids[i]));
+            }
+        }
+    }
+
+    private:
+    Argument& arg;
+    std::size_t iter           = 0;
+    std::size_t rotating_count = 1;
+    std::size_t size_a         = 0;
+    std::size_t size_b         = 0;
+    std::size_t size_c         = 0;
+    std::vector<const void*> p_a_grids;
+    std::vector<const void*> p_b_grids;
+    std::vector<void*> p_c_grids;
+};
+
+static void flush_icache()
+{
+    hipDeviceProp_t deviceProps;
+    hip_check_error(hipGetDeviceProperties(&deviceProps, 0));
+    int32_t gpu_block3 = deviceProps.multiProcessorCount * 60;
+
+    int flush_iter = 20000;
+
+    for(int i = 0; i < flush_iter; i++)
+    {
+        ck::flush_icache<<<dim3(gpu_block3), dim3(64), 0, nullptr>>>();
+    }
```

Comment:
As we just want to refresh the I$ by issuing a lot of s_icache_inv + s_nops, we'd better to move flush_iter inside gpu kernel because we save many kernel launch latency when use ckProfiler. e.g.

1.  Current implementation: launch (s_icache_inv  + 14 s_nops) kernel 20000 times
2.  Proposed implementation: launch ( 100 *(s_icache_inv  + 14 s_nops)) kernel 200 times

2 will save 99% time of 1, as kernel launch time dominate these very small kernel overhead.